libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -512,6 +521,35 @@ bool AP_BattMonitor::get_cycle_count(uint8_t instance, uint16_t &cycles) const
     return drivers[instance]->get_cycle_count(cycles);
 }
 
+bool AP_BattMonitor::get_serial_number(uint8_t instance, char *serial_number, uint8_t buflen) const
+{
+    if (instance >= AP_BATT_MONITOR_MAX_INSTANCES || (drivers[instance] == nullptr)) {
+        return false;
+    }
+    return drivers[instance]->get_serial_number(serial_number, buflen);
+}
+
+bool AP_BattMonitor::get_product_name(uint8_t instance, char *product_name, uint8_t buflen) const
+{
+    if (instance >= AP_BATT_MONITOR_MAX_INSTANCES || (drivers[instance] == nullptr)) {
+        return false;
+    }
+
+    char manufacturer_name[SMBUS_READ_BLOCK_MAXIMUM_TRANSFER+1];
```

Comment:
Frontend should probably not know about SMBus.

But not a stopper for me, at least.

Reply:
Done. Moved this to backend.